tools/tokenserver/util.py

```diff
@@ -23,26 +28,59 @@ def run_script(main):
     sys.exit(exitcode)
 
 
-def configure_script_logging(opts=None):
+def configure_script_logging(opts=None, logger_name=""):
     """Configure stdlib logging to produce output from the script.
 
     This basically configures logging to send messages to stderr, with
     formatting that's more for human readability than machine parsing.
     It also takes care of the --verbosity command-line option.
     """
-    if not opts or not opts.verbosity:
-        loglevel = logging.WARNING
-    elif opts.verbosity == 1:
-        loglevel = logging.INFO
+
+    verbosity = (
+        opts and getattr(
+            opts, "verbosity", logging.NOTSET)) or logging.NOTSET
+    logger = logging.getLogger(logger_name)
+    level = os.environ.get("PYTHON_LOG", "").upper() or \
+        max(logging.DEBUG, logging.WARNING - (verbosity * 10)) or \
+        logger.getEffectiveLevel()
+
+    # if we've already written a log message, the handler may already
+    # be defined. We don't want to duplicate messages if possible, so
+    # check and potentially adjust the existing logger's handler.
```

Comment:
Sorry, one more thing: is this still a problem?

I added a logging call prior to `configure_script_logging` and the root logger still had an empty `handlers` here, which is what I would expect. Something like `basicConfig` might adjust the handlers but not a stray logging call by itself?

Reply:
Not sure I understand that use case (could just call `configure_script_logging` earlier especially w/ it requiring no args, or just use print) but fair enough, I think the comment could be clarified then since it's not about previous log statements, e.g.
```suggestion
    # if we've previously setup a handler, adjust it instead
```